### PR TITLE
Remove network lock

### PR DESCRIPTION
**Pull request by AgeManning**
## Description

This removes the lock around the network service. 

Specific variables now have their own locks in a struct called `NetworkGlobals`. 

I've moved some more logic into the main network thread to accommodate this.

### CODE REVIEW

**Comment by AgeManning** [comment]
@pawanjay176 - I've moved your DHT persistent tests and they are currently commented out. We no longer have access to the network service from outside the main network thread. 

I'm not if you can re-jig these tests to make them work in the new layout or not?

File: beacon_node/eth2-libp2p/src/discovery.rs [hunk 1]
```diff
@@ -57,6 +55,10 @@ pub struct Discovery<TSubstream> {
     /// The discovery behaviour used to discover new peers.
     discovery: Discv5<TSubstream>,
 
+    /// A collection of network constants that can be read from other threads.
+    /// behaviour.
```
Comment (pawanjay176) [inline beacon_node/eth2-libp2p/src/discovery.rs]: behaviour seems out of place.

File: beacon_node/network/src/service.rs [hunk 2]
```diff
@@ -40,48 +42,42 @@ impl<T: BeaconChainTypes> Service<T> {
     ) -> error::Result<(Arc<Self>, mpsc::UnboundedSender<NetworkMessage>)> {
         // build the network channel
         let (network_send, network_recv) = mpsc::unbounded_channel::<NetworkMessage>();
-        // Get a reference to the beacon chain store
-        let store = beacon_chain.store.clone();
         // launch message handler thread
+        let store = beacon_chain.store.clone();
```
Comment (pawanjay176) [inline beacon_node/network/src/service.rs]: switch lines

File: beacon_node/network/src/service.rs [hunk 3]
```diff
@@ -40,48 +42,42 @@ impl<T: BeaconChainTypes> Service<T> {
...
-        };
-        for enr in enrs {
-            libp2p_service.lock().swarm.add_enr(enr);
+        for enr in load_dht::<T>(store.clone()) {
+            libp2p_service.swarm.add_enr(enr);
         }
 
-        let libp2p_exit = spawn_service(
-            libp2p_service.clone(),
+        // a delay used to initialise code after the network has started
```
Comment (pawanjay176) [inline beacon_node/network/src/service.rs]: This comment should be a little more descriptive as to why we need the delay imo

File: beacon_node/network/src/service.rs [hunk 4]
```diff
@@ -40,48 +42,42 @@ impl<T: BeaconChainTypes> Service<T> {
...
+        let initial_delay = Delay::new(Instant::now() + Duration::from_secs(1));
```
Comment (pawanjay176) [inline beacon_node/network/src/service.rs]: Delay duration could be a const value.

**Comment by AgeManning** [comment]
Would it not be possible to start a network service with a collection of bootnodes, shutdown the service and then run `load_dht()` on the store to see if the they were persisted?

**Comment by pawanjay176** [comment]
@AgeManning yeah you are right. Didn't think of loading directly from the store.

**Pull request merged**